src/roslibpy/actionlib.py

```diff
@@ -428,14 +432,26 @@ def set_succeeded(self, result):
         LOGGER.info("Action server {} setting current goal to SUCCEEDED".format(self.server_name))
 
         with self._lock:
+
+            self.status_message["status_list"] = [
+                dict(
+                    goal_id=self.current_goal["goal_id"],
+                    status=GoalStatus.SUCCEEDED)
+            ]
+            self._publish_status()
```

Comment:
Nice fix!

Should there maybe be a call to `_publish_status()` in `_on_goal_message` as well?

This has little effect since the local status of the goal at the client side is set to `PENDING` anyways, but nevertheless for short enough Actions (like the Fibonacci example) the status is set but never sent, it's being reset upon completion.